pkg/api/validate/vm.go

```diff
@@ -7,66 +7,90 @@ import (
+	api.VMSizeStandardD32sV3:  true,
+	// Memory optimized
+	api.VMSizeStandardE4sV3:     true,
+	api.VMSizeStandardE8sV3:     true,
+	api.VMSizeStandardE16sV3:    true,
+	api.VMSizeStandardE32sV3:    true,
+	api.VMSizeStandardE64isV3:   true,
+	api.VMSizeStandardE64iV3:    true,
+	api.VMSizeStandardE80isV4:   true,
+	api.VMSizeStandardE80idsV4:  true,
+	api.VMSizeStandardE104iV5:   true,
+	api.VMSizeStandardE104isV5:  true,
+	api.VMSizeStandardE104idV5:  true,
+	api.VMSizeStandardE104idsV5: true,
+	// Compute optimized
+	api.VMSizeStandardF4sV2:  true,
+	api.VMSizeStandardF8sV2:  true,
+	api.VMSizeStandardF16sV2: true,
+	api.VMSizeStandardF32sV2: true,
+	api.VMSizeStandardF72sV2: true,
+	// Memory and storage optimized
+	api.VMSizeStandardG5:  true,
+	api.VMSizeStandardGS5: true,
+	// Memory and compute optimized
+	api.VMSizeStandardM128ms: true,
+	// Storage optimized
+	api.VMSizeStandardL4s:    true,
+	api.VMSizeStandardL8s:    true,
+	api.VMSizeStandardL16s:   true,
+	api.VMSizeStandardL32s:   true,
+	api.VMSizeStandardL8sV2:  true,
+	api.VMSizeStandardL16sV2: true,
+	api.VMSizeStandardL32sV2: true,
+	api.VMSizeStandardL48sV2: true,
+	api.VMSizeStandardL64sV2: true,
+}
+
 func DiskSizeIsValid(sizeGB int) bool {
 	return sizeGB >= 128
 }
 
 func VMSizeIsValid(vmSize api.VMSize, requiredD2sV3Workers, isMaster bool) bool {
 	if isMaster {
-		switch vmSize {
-		case api.VMSizeStandardD8sV3,
-			api.VMSizeStandardD16sV3,
-			api.VMSizeStandardD32sV3,
-			api.VMSizeStandardE64iV3,
-			api.VMSizeStandardE64isV3,
-			api.VMSizeStandardF72sV2,
-			api.VMSizeStandardGS5,
-			api.VMSizeStandardG5,
-			api.VMSizeStandardM128ms:
+		if supportedMasterVMSizes[vmSize] {
 			return true
 		}
-	} else {
-		if requiredD2sV3Workers {
-			switch vmSize {
-			case api.VMSizeStandardD2sV3:
-				return true
-			}
-		} else {
-			switch vmSize {
-			case api.VMSizeStandardD4asV4,
-				api.VMSizeStandardD8asV4,
-				api.VMSizeStandardD16asV4,
-				api.VMSizeStandardD32asV4,
-				api.VMSizeStandardD4sV3,
-				api.VMSizeStandardD8sV3,
-				api.VMSizeStandardD16sV3,
-				api.VMSizeStandardD32sV3,
-				api.VMSizeStandardE4sV3,
-				api.VMSizeStandardE8sV3,
-				api.VMSizeStandardE16sV3,
-				api.VMSizeStandardE32sV3,
-				api.VMSizeStandardE64iV3,
-				api.VMSizeStandardE64isV3,
-				api.VMSizeStandardF4sV2,
-				api.VMSizeStandardF8sV2,
-				api.VMSizeStandardF16sV2,
-				api.VMSizeStandardF32sV2,
-				api.VMSizeStandardF72sV2,
-				api.VMSizeStandardG5,
-				api.VMSizeStandardGS5,
-				api.VMSizeStandardM128ms,
-				api.VMSizeStandardL4s,
-				api.VMSizeStandardL8s,
-				api.VMSizeStandardL16s,
-				api.VMSizeStandardL32s,
-				api.VMSizeStandardL8sV2,
-				api.VMSizeStandardL16sV2,
-				api.VMSizeStandardL32sV2,
-				api.VMSizeStandardL48sV2,
-				api.VMSizeStandardL64sV2:
-				return true
-			}
-		}
 	}
+
+	if supportedWorkerVMSizes[vmSize] && !requiredD2sV3Workers ||
+		requiredD2sV3Workers && vmSize == api.VMSizeStandardD2sV3 {
+		return true
+	}
```

Comment:
Now that I look at it, I'm pretty sure this condition does not do what you expect because it is lacking parenthesis. 
```suggestion
	if (supportedWorkerVMSizes[vmSize] && !requiredD2sV3Workers) ||
		(requiredD2sV3Workers && vmSize == api.VMSizeStandardD2sV3) {
		return true
	}
```

Reply:
I'm not sure parentheses are needed, but for readability, we could go with something like this instead:
```
if supportedWorkerVMSizes[vmSize] && !requiredD2sV3Workers {
     return true
}

if requiredD2sV3Workers && vmSize == api.VMSizeStandardD2sV3 {
     return true
}
```
What do you think?